### PR TITLE
Add feature page for following feature: WebAssembly Extensions [Alpha]

### PR DESCRIPTION
**Pull request by johnma14**

### CODE REVIEW

File: features/webassembly_extensions.md [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+
+- [ ] Explicit user action is required to enable this feature (e.g. a config field, config resource, or installation action). 
+
+> Link to instructions for enabling
+
+**Docs**
+
+- [ ] Reference docs are published to preliminary.istio.io or the Istio wiki.
+- [X] Basic feature docs are published on preliminary.istio.io describing what the feature does, how to use it, and any caveats.
+  [Extensibility Concepts] (https://istio.io/latest/docs/concepts/wasm/)
```
Comment (brian-avery) [inline features/webassembly_extensions.md:92]:
```suggestion
  [Extensibility Concepts] (https://istio.io/latest/docs/concepts/wasm/) 
  [Configuration](https://istio.io/latest/docs/ops/configuration/extensibility/wasm-module-distribution/)
```

**Comment by brian-avery** [comment]
@istio/wg-networking-maintainers can you please review this?

**Comment by brian-avery** [comment]
@mandarjog gentle reminder. Can you please review?

File: features/webassembly_extensions.md [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+### Requirements: 
+
+**Design**
+
+- [X] RFC has been approved describing the intention of the feature as well as the user stories behind the feature. 
+
+ [WebAssembly Runtime Validation](https://docs.google.com/document/d/12WUe09imQf6gGgbOZN_vQOcKkd_nek3v7-aGHdNeeSU/edit#)
+ [Istio WebAssembly Plugin SDK](https://docs.google.com/document/d/1_MUmeaUxuL1Ialr_Dvk7vYchDp4GFAtHrLBJnBzODcs/edit#heading=h.xw1gqgyqs5b)
+ [Wasm Extensions API](https://docs.google.com/document/d/1JKg2V86WUftNs5uYCuQ-9X4-JVyF-jM5dpfDyodm5r0/edit#heading=h.xw1gqgyqs5b)
+ [Istio Extensions Configuration API](https://docs.google.com/document/d/1jFuZM4UCKw6AMDmXMIYqbdlyJk95ut2zMPLSNdrUtW8/edit#heading=h.qex63c29z2to)
```
[the remark this answers is not in the document]
Comment (bianpengyuan) [reply features/webassembly_extensions.md:83]: Also seeing something weird in the page: `buf:lint:ignore ENUMVALUEUPPERSNAKECASE`..

File: features/webassembly_extensions.md [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+
+**Config**
+
+- [X] Explicit user action is required to enable this feature (e.g. a config field, config resource, or installation action). 
+[Istio Ecosystem Wasm Extensions](https://github.com/istio-ecosystem/wasm-extensions)
+> Link to instructions for enabling
+
+**Docs**
+
+- [ ] Reference docs are published to preliminary.istio.io or the Istio wiki.
```
Comment (bianpengyuan) [inline features/webassembly_extensions.md:93]: I will try to get the existing Wasm doc updated to reflect the new API this week.

**Review by bianpengyuan — COMMENTED** [review]
lgtm

**Comment by johnma14** [comment]
Thank you @mandarjog & @bianpengyuan. Made the recommended changes.

**Pull request closed without merging**